src/lib/Syntax.hs

```diff
@@ -245,18 +246,21 @@ data UExpr' = UVar UVar
               deriving (Show, Generic)
 
 data UConDef = UConDef Name (Nest UAnnBinder)  deriving (Show, Generic)
-data UDecl = ULet LetAnn UPatAnn UExpr
-           | UData UConDef [UConDef]
-           | UInterface [UType] UConDef [UAnnBinder]
-           | UInstance UType [(UVar, UExpr)]
-             deriving (Show, Generic)
+data UDecl =
+   ULet LetAnn UPatAnn UExpr
+ | UData UConDef [UConDef]
+ | UInterface [UType] UConDef [UAnnBinder] -- superclasses, constructor, methods
+ | UInstance (Nest UPatAnnArrow) UType [UMethodDef]  -- args, type, methods
```

Comment:
What are "args"? Is this a list of binders to prepend to each method definition?